src/container.ts

```diff
@@ -536,6 +551,38 @@ export class Container {
       });
     }
 
+    // Resource requests and limits
```

Comment:
This feels a little cluttered...can you try making the flow clearer? maybe something like:

```ts
    const cpuLimit = this.resources?.cpu?.limit?.amount;
    const cpuRequest = this.resources?.cpu?.request?.amount;
    const memoryLimit = this.resources?.memory?.limit;
    const memoryRequest = this.resources?.memory?.request;

    const limits: { [key: string]: k8s.Quantity } = {};
    const requests: { [key: string]: k8s.Quantity } = {};

    if (cpuLimit) {
      limits.cpu = k8s.Quantity.fromString(cpuLimit);
    }
    if (memoryLimit) {
      limits.memory = k8s.Quantity.fromString(memoryLimit.toMebibytes().toString() + 'Mi');
    }
    if (cpuRequest) {
      requests.cpu = k8s.Quantity.fromString(cpuRequest);
    }
    if (memoryRequest) {
      requests.memory = k8s.Quantity.fromString(memoryRequest.toMebibytes().toString() + 'Mi');
    }

    const resourceRequirements = {
      limits: limits,
      requests: requests,
    };
```

?

Reply:
Thanks @iliapolo. I just committed the changes above except for the following block:
```
readonly      let resourceRequirements: ResourceRequirements | undefined = undefined;
    if (Object.keys(limits).length > 0 || Object.keys(requests).length > 0) {
      resourceRequirements = {
        limits: limits,
        requests: requests,
      };
    }
```
If I don't allow `resourceRequirements` to be undefined, it ends up being an "empty" object, and then the 'Instantiation properties are all respected' test fails.

Reply:
Fair enough 👍